infra/src/main/kotlin/fi/fta/geoviite/infra/tracklayout/LayoutGeometry.kt

```diff
@@ -166,23 +166,8 @@ interface IAlignment : Loggable {
         .minByOrNull { (distance, _) -> distance }
         ?.let { (_, index) -> index }
 
-    fun getMaxDirectionDeltaRads(): Double {
-        var prevPoint: SegmentPoint? = null
-        var prevDirection: Double? = null
-        return if (segments.isEmpty()) {
-            0.0
-        } else {
-            segments.maxOf { segment ->
-                segment.segmentPoints.maxOf { point ->
-                    val direction = prevPoint?.let { prev -> directionBetweenPoints(prev, point) }
-                    angleDeltaRads(prevDirection, direction).also {
-                        prevPoint = point
-                        prevDirection = direction
-                    }
-                }
-            }
-        }
-    }
+    fun getMaxDirectionDeltaRads(): Double =
+        allSegmentPoints.zipWithNext(::directionBetweenPoints).zipWithNext(::angleDeltaRads).maxOrNull() ?: 0.0
```

Comment:
Tuo aiempi bugasi ilmeisesti koska vertasi edellisen segmentin viimeistä seuraavan ensimmäiseen pisteeseen ja ne on oikeasti samat jolloin kulma menee jojoon? Korjaus ihan jees ja onhan tuo nyt kaunis noin :+1: 

Syy miksi se oli tehty noin oli että in-place iterointi vältti tekemästä uusia listoja koska isolla alignmentilla on paljon pisteitä. Nythän tuo tekee 1) uuden listan kaikista pisteistä 2) uuden listan pistepareista 3) uuden listan kulma-arvoista ja sitten kaivaa siitä isoimman. Ei ehkä mitenkään kriittinen juttu ja tämä on joka tapauksessa parempi kuin rikkinäinen edeltäjä. 

Silti, tänne voisi harkita tekevänsä jonkunlaiset in-place iterointifunkkarit tyyliin `mapPoints(funktio: SegmentPoint -> T)`, `reducePoints(funktio: (T, SegmentPoint) -> T)` jne, joilla saisi nämä toteutettua vähemmällä listanluonnilla.

Reply:
En alkuaan viitsinyt lähteä noita säätämään siksi, että saisi vältettyä jonkin pari millisekuntia elävän parinkymmenen megatavun allokaation, mutta äkkiähän nyt siitäkin pääsee eroon. Se omana pullarinaan https://github.com/finnishtransportagency/geoviite/pull/931.

Reply:
Joo eihän se maata merelle tuo, mutta ehkä juuri sen verran aktiivinen että luo turhaa roskienkeruupainetta, varsinkin jos summaantuu muiden vastaavien käyttöpaikkojen kanssa. Tuo olikin vielä simppelimpi ratkaisu :+1: